docs/_pages/typecheck.md

```diff
@@ -287,6 +287,64 @@ In type annotations, this is written as `...T`:
 type F = (...number) -> ...string
 ```
 
+## Type packs
+
+Multiple function return values as well as the function variadic parameter uses a type pack to represent a list of types.
+
+When a type alias is defined, generic type pack parameters can be used after the type parameters:
+
+```lua
+type Signal<T, U...> = { f: (T, U...) -> (), data: T }
+```
+
+Similarly, generic function can reference a generic type pack from the generics list:
```

Comment:
Can we add a note before this that explains that `...T` means "arbitrary number of values of type T" whereas `T...` means "a number of values that match types defined in T... type pack". It may be redundant but it's a common point of confusion.

Reply:
Added a note.